flink-connectors/flink-connector-pulsar/src/main/java/org/apache/flink/connector/pulsar/sink/writer/topic/TopicProducerRegister.java

```diff
@@ -199,4 +203,13 @@ private void abortTransactions() {
     private void clearTransactions() {
         transactionRegister.clear();
     }
+
+    /** Retrieve the producer metrics and map to Flink SinkWriterMetrics. */
+    public void updateProducerStats() {
+        pulsarSinkWriterMetrics.updateProducerStats(producerRegister);
```

Comment:
currently the producer register impl has changed, wondering in what case the a topic will have different schemaInfo (thus different producers) or we just want to associate each producer with a schemaInfo ? Thanks ~